trendly/src/app/histogram-section/histogram-section.component.ts

```diff
@@ -1,136 +1,157 @@
-    this.convertDataToChartsFormat();
+    // this function should be call after data retrival from server, will change
+    // after creating backend this.convertDataToChartsFormat();
   }
 
   /**
    * converts the data from the server to charts format.
    */
   convertDataToChartsFormat() {
-    const topics : Map<string, number> = this.extractTopics(MOCK_DATA);
+    const topics: Map<string, number> = this.extractTopics(this.trendsData);
     this.createColumnNames(topics);
     this.createData(topics);
   }
 
   /**
    * extracts and returns the topics maps to its' index.
    */
-  extractTopics(data: Object) :Map<string, number> {
-    const topics : Map<string, number> = new Map<string, number>();
-    let counter : number = 0;
+  extractTopics(data: Object): Map<string, number> {
+    const topics: Map<string, number> = new Map<string, number>();
+    let counter: number = 0;
     for (let key of Object.keys(data)) {
       for (let element of data[key]) {
-        if (!topics.has(element.name))
-        {
+        if (!topics.has(element.name)) {
           topics.set(element.name, counter);
           counter++;
         }
       }
     }
     return topics;
-   }
-
-   /**
-    * creates the columns forthe chart.
-    * @param topics 
-    */
-   createColumnNames(topics : Map<string, number>) : void {
-      this.columnNames[0] = 'Topic';
-      let index :number = 0;
-     for (let key of topics.keys()) {
-       index = topics.get(key) * 3;
-       this.columnNames[index + 1] = key;
-       this.columnNames[index+ 2] = {role:TOOLTIP_ROLE};
-       this.columnNames[index + 3] = {role: STYLE_ROLE};
-     }
-   }
-
-   /**
-    * creates the data for the charts.
-    * @param topics
-    */
-   private createData(topics : Map<string, number>) : void{
-     for (let date of Object.keys(MOCK_DATA)) {
-       const row = Array((topics.size * 3) + 1).fill('');
-       this.initializeRowArray(row);
-       row[0] = date;
-
-       for (let element of MOCK_DATA[date]) {
-         const index = topics.get(element.name) * 3;
-         const indexColor = topics.get(element.name);
-         row[index + 1] = element.volume;
-         row[index + 2] = element.description;
-         row[index + 3] = this.coloresService._lightColorShow[indexColor];
-       }
-       this.data.push(row);
-     }
-   }
-
-   /**
-    * initializes one row in the data.
-    * @param array
-    */
-  initializeRowArray(array : Array<string | number>) : void {
-    for (let i = 1; i < array.length; i+=3) {
+  }
+
+  /**
+   * creates the columns forthe chart.
+   * @param topics
+   */
+  createColumnNames(topics: Map<string, number>): void {
+    this.columnNames = [];
+    this.columnNames[0] = 'Topic';
+    let index: number = 0;
+    for (let key of topics.keys()) {
+      index = topics.get(key) * 3;
+      this.columnNames[index + 1] = key;
+      this.columnNames[index + 2] = {role: TOOLTIP_ROLE};
+      this.columnNames[index + 3] = {role: STYLE_ROLE};
+    }
```

Comment:
Tip: use Map.entries() together with destructuring assignment
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
as follows:
```suggestion
    for (const [key, i] of topics.entries()) {
      let index = i*3;
      this.columnNames[++index] = key;
      this.columnNames[++index] = {role: TOOLTIP_ROLE};
      this.columnNames[++index] = {role: STYLE_ROLE};
    }
```

Reply:
cool, will modify it.